### ISSUE TITLE
Adding validation set evaluation for reranker finetuning

### ISSUE THREAD

**Issue by jackkwok** [FlagOpen/FlagEmbedding#838]
To avoid overfitting, I would like to add validation (eval) dataset to the finetuning.  I see the BiTrainer is a subclass of Huggingface's [Trainer](https://huggingface.co/docs/transformers/en/main_classes/trainer) class which accepts a eval_dataset argument.  I plan to add the eval dataset within the initialization of BiTrainer ([line](https://github.com/FlagOpen/FlagEmbedding/blob/master/FlagEmbedding/llm_reranker/finetune_for_instruction/run.py#L102)). 

Let me know if there are other considerations.

**Comment by EEEnterURname** [comment]
@jackkwok Did you successfully add the validation dataset?If successful, can you share how to achieve it?
